pad.c

```diff
@@ -982,8 +986,12 @@ Perl_pad_findmy_pvn(pTHX_ const char *namepv, STRLEN namelen, U32 flags)
 
     offset = pad_findlex(namepv, namelen, flags,
                 PL_compcv, PL_cop_seqmax, 1, NULL, &out_pn, &out_flags);
-    if (offset != NOT_IN_PAD)
+    if (offset != NOT_IN_PAD) {
+        /* is this a lexical import that has been deleted? */
+        if (PadnameIsTOMBSTONE(out_pn))
+            return NOT_IN_PAD;
```

Comment:
Just wondering why this check occurs here, rather than in `S_pad_findlex`. Is that latter function essentially doing unnecessary work when `PadnameIsTOMBSTONE` is true and could bail out earlier if the check was implemented there instead?

Reply:
An excellent question. As it turns out it would be better to put it in `S_pad_findlex` so as not to perform additional outer captures of tombstones or other weirdness.

I've edited it and force-pushed.

Reply:
Cool, I don't have any more comments.